zerver/lib/actions.py

```diff
@@ -4394,20 +4394,29 @@ def get_email_gateway_message_string_from_address(address: str) -> Optional[str]
 
     return msg_string
 
-def decode_email_address(email: str) -> Optional[Tuple[str, str]]:
-    # Perform the reverse of encode_email_address. Returns a tuple of (streamname, email_token)
+def decode_email_address(email: str) -> Optional[Tuple[str, str, bool]]:
+    # Perform the reverse of encode_email_address. Returns a tuple of
+    # (streamname, email_token, show_sender)
     msg_string = get_email_gateway_message_string_from_address(email)
+    show_sender = False
 
     if msg_string is None:
         return None
     elif '.' in msg_string:
         # Workaround for Google Groups and other programs that don't accept emails
         # that have + signs in them (see Trac #2102)
-        encoded_stream_name, token = msg_string.split('.')
+        parts = msg_string.split('.')
```

Comment:
I think might be cleaner (for future people reading this code) to just check that `msg_string.endswith('show-sender')` on line 4 of this function, and then strip it.